analyzer/src/main/kotlin/managers/Npm.kt

```diff
@@ -301,7 +301,10 @@ open class Npm(
 
         log.info { "Searching for 'package.json' files in '$nodeModulesDir'..." }
 
-        val nodeModulesFiles = nodeModulesDir.walk().filter {
+        // PNPM symlinks workspace modules in the `node_modules` directory, which will result in cyclic symlinks of
+        // `node_module` directories. Limit the search depth to '2' in this case.
```

Comment:
Why is `2` a reasonable search depth? e.g. why doesn't it miss anything?

Reply:
Because the `pnpm install` option `--shamefully-hoist` lifts all dependencies to the top level `node_modules` directory.
That means that all `package.json` files will be located in a direct subdirectory of `node_modules` which results in a required search depth of `2` in the case of `PNPM`.

Reply:
Could you please update the code comment to reflect that?

Reply:
Did that in the newly introduced method.